### PR TITLE
firebase-get-to-know-flutter: add go_router

### PR DESCRIPTION
**Pull request by miquelbeltran**
TODO: Rebase with main once https://github.com/flutter/codelabs/pull/1206 lands.

In this PR I continue the work on migrating the firebase-get-to-know codelab to go_router.

The work is in the commits with the title "Step xx: add go_router"

## Pre-launch Checklist

- [x] I read the [Effective Dart: Style] _recently_, and have followed its advice.
- [x] I signed the [CLA].
- [ ] I updated/added relevant documentation (doc comments with `///`). <-- needs codelab update
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-devrel channel on [Discord].

[Effective Dart: Style]: https://dart.dev/guides/language/effective-dart/style
[CLA]: https://cla.developers.google.com/
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by domesticmouse** [comment]
Code formatting required

```console
== Testing
Analyzing step_02...
No issues found!
Changed lib/src/authentication.dart
Formatted 5 files (1 changed) in 0.27 seconds.
```

**Comment by miquelbeltran** [comment]
Code formatted, although waiting for #1206 to land to rebase it

**Comment by domesticmouse** [comment]
> Code formatted, although waiting for #1206 to land to rebase it

rebase away!

File: firebase-get-to-know-flutter/step_06/lib/main.dart [hunk 1]
```diff
@@ -4,83 +4,103 @@
...
+import 'package:go_router/go_router.dart';
 import 'package:google_fonts/google_fonts.dart';
 import 'package:provider/provider.dart';
 
 import 'app_state.dart';
 import 'home_page.dart';
 
 void main() {
+  WidgetsFlutterBinding.ensureInitialized();
+
```
Comment (miquelbeltran) [inline firebase-get-to-know-flutter/step_06/lib/main.dart:16]: This was missing in the Step 06, but it is present on all other steps

**Comment by domesticmouse** [comment]
Please pull latest from main, as the CI failures should be fixed.

File: firebase-get-to-know-flutter/step_06/lib/main.dart (hunk 1 above)
Comment (domesticmouse) [reply firebase-get-to-know-flutter/step_06/lib/main.dart:16]: sgtm

**Review by domesticmouse — APPROVED** [review]

**Comment by domesticmouse** [comment]
@miquelbeltran is this ready to land?

**Comment by miquelbeltran** [comment]
Yes, good to go

**Pull request merged**